expressjs-backend/routes/listingRoutes.js

```diff
@@ -75,7 +75,7 @@ router.get("/", async (req, res) => {
 
     // If there is a search query, modify the SQL query to include a WHERE clause
     if (q) {
-      query += ` WHERE "title" LIKE '%${q}%' OR "description" LIKE '%${q}%'`;
+      query += ` WHERE "title" ILIKE '%${q}%' OR "description" ILIKE '%${q}%' OR "category"::text ILIKE '%${q}%'`;
```

Comment:
check if there is a search query and then build the query entirely. if there isn't a query we can improve performance by removing the additional statement. what I mean is query = ' SELECT * FROM "table" WHERE "title" ILIKE '%${q}%' OR "description" ILIKE '%${q}%' OR "category"::text ILIKE '%${q}%'`; as opposed to concatenating